scipy/special/tests/test_basic.py

```diff
@@ -2715,9 +2720,26 @@ def test_rgamma(self):
         assert_almost_equal(rgam,rlgam,8)
 
     def test_infinity(self):
-        assert_(np.isinf(special.gamma(-1)))
         assert_equal(special.rgamma(-1), 0)
 
+    @pytest.mark.parametrize(
+        "x,expected",
+        [
+            # infinities
+            ([-np.inf, np.inf], [np.nan, np.inf]),
+            # zeros
+            ([-0.0, 0.0], [-np.inf, np.inf]),
```

Comment:
Ah, good point about "zeros".  I meant two "zeros" for input as in `+0.0` and `-0.0`, but then it makes it sound like I'm talking about zeros of the zeroless gamma function. I'll change the comment to something like `# positive and negative zero`

The second pair here isn't actually repeated by the way, the format is that it's a tuple, of input value `x`, and the expected reference value. It's not obvious though. Because these are annoying to read due to the context for what each element of the tuple means (here `"x,expected"`) being far from the tuples, I actually used named tuples for the `hyp2f1` test suite:

https://github.com/scipy/scipy/blob/3367349b6bea4742c0ab751ba1a624303707707a/scipy/special/tests/test_hyp2f1.py#L79-L85

which makes it a little more clear what's going on. I could do that here, but it might be obstrusive to add to `test_basic.py`.

Reply:
> The second pair here isn't actually repeated by the way, the format is that it's a tuple, of input value `x`, and the expected reference value. It's not obvious though.

Indeed I missed that both `x` and `expected` are lists here.